lua/telescope/pickers.lua

```diff
@@ -401,18 +405,31 @@ function Picker:find()
       end
 
       -- TODO: Entry manager should have a "bulk" setter. This can prevent a lot of redraws from display
-      self.manager = EntryManager:new(self.max_results, self.entry_adder, self.stats)
+      if not self.cached_picker then
+        self.manager = EntryManager:new(self.max_results, self.entry_adder, self.stats)
 
-      local find_id = self:_next_find_id()
-      local process_result = self:get_result_processor(find_id, prompt, debounced_status)
-      local process_complete = self:get_result_completor(self.results_bufnr, find_id, prompt, status_updater)
+        local find_id = self:_next_find_id()
+        local process_result = self:get_result_processor(find_id, prompt, debounced_status)
+        local process_complete = self:get_result_completor(self.results_bufnr, find_id, prompt, status_updater)
 
-      local ok, msg = pcall(function()
-        self.finder(prompt, process_result, vim.schedule_wrap(process_complete))
-      end)
+        local ok, msg = pcall(function()
+          self.finder(prompt, process_result, vim.schedule_wrap(process_complete))
+        end)
 
-      if not ok then
-        log.warn("Failed with msg: ", msg)
+        if not ok then
+          log.warn("Failed with msg: ", msg)
+        end
+      else
+        -- resume previous picker
+        local index = 1
+        for entry in self.manager:iter() do
+          self:entry_adder(index, entry, _, true)
+          index = index + 1
+        end
+        self.cached_picker = false
+        -- TODO(fdschmidt93) this does not work for live_grep
+        -- redraw highlighting for multi-selections
+        self.refresh()
```

Comment:
I've been dreading already too long over this.

The straightforward approach of 

```lua
local index = 1
for entry in self.manager:iter() do
  self:entry_adder(index, entry, _, true)
  index = index + 1
  if self:can_select_row(row) and self._multi:is_selected(entry) then
    self.highlighter:hi_multiselect(row, true)
  end
end
```

doesn't work. I've tried a lot of other hacks I could think of but to no avail. In any case, selected entries in any case would be highlighted once you scroll over them.